instrumentation/spring/spring-boot-autoconfigure/src/main/java/io/opentelemetry/instrumentation/spring/autoconfigure/internal/properties/SpringConfigProperties.java

```diff
@@ -29,37 +30,86 @@ public class SpringConfigProperties implements ConfigProperties {
   private final ExpressionParser parser;
   private final OtlpExporterProperties otlpExporterProperties;
   private final OtelResourceProperties resourceProperties;
-  private final PropagationProperties propagationProperties;
   private final ConfigProperties otelSdkProperties;
+  private final ConfigProperties customizedListProperties;
+  private final Map<String, List<String>> listPropertyValues = new HashMap<>();
+
+  static final String DISABLED_KEY = "otel.java.disabled.resource.providers";
+  static final String ENABLED_KEY = "otel.java.enabled.resource.providers";
 
   public SpringConfigProperties(
       Environment environment,
       ExpressionParser parser,
       OtlpExporterProperties otlpExporterProperties,
       OtelResourceProperties resourceProperties,
-      PropagationProperties propagationProperties,
+      OtelSpringProperties otelSpringProperties,
       ConfigProperties otelSdkProperties) {
     this.environment = environment;
     this.parser = parser;
     this.otlpExporterProperties = otlpExporterProperties;
     this.resourceProperties = resourceProperties;
-    this.propagationProperties = propagationProperties;
     this.otelSdkProperties = otelSdkProperties;
+    this.customizedListProperties =
+        createCustomizedListProperties(otelSdkProperties, otelSpringProperties);
+
+    listPropertyValues.put(ENABLED_KEY, otelSpringProperties.getJavaEnabledResourceProviders());
+    listPropertyValues.put(DISABLED_KEY, otelSpringProperties.getJavaDisabledResourceProviders());
+    listPropertyValues.put(
+        "otel.experimental.metrics.view.config",
+        otelSpringProperties.getExperimentalMetricsViewConfig());
+    listPropertyValues.put(
+        "otel.experimental.resource.disabled.keys",
+        otelSpringProperties.getExperimentalResourceDisabledKeys());
+    listPropertyValues.put("otel.propagators", otelSpringProperties.getPropagators());
+  }
+
+  private static Map<String, String> addList(
```

Comment:
The method implementation has changed. The method name should perhaps reflect the new implementatiion? Perhaps something like `createMapForListProperty`?